internal/tiltfile/k8s.go

```diff
@@ -632,23 +632,35 @@ func makeWorkloadToResourceFunction(f *starlark.Function) (workloadToResourceFun
 	}, nil
 }
 
-func (s *tiltfileState) checkResourceConflict(name string) error {
-	if s.k8sByName[name] != nil {
-		return fmt.Errorf("k8s_resource named %q already exists", name)
+func (s *tiltfileState) checkResourceConflicts() error {
```

Comment:
My big high-level concern with this approach is that users don't understand how resource names are generated. 

The old approach deliberately pointed them at the line of code where the resource name was generated. 

The new approach doesn't give them any hints. I don't think users will know how to fix this new error and we will get lots of complaints about this.

Reply:
Yeah, doing the old approach is trickier now with the ability to rename both k8s and dc resources. And it shows up in a worse errors experience. I'll give another go at improving it. Maybe it will work to track the files and line numbers of all the resources as they are created and then use them here to report errors during conflict resolution.

Reply:
for now, could we go back to the old way of checking resource name conflicts on creation rather than on assembly?

i understand that the assembly approach makes it possible to rename resources and move things around...but i don't think we actually have the debugging tools to support a good user experience for that.

Reply:
(i.e., i don't think we want users to have to think really hard about the order that resource naming happens in. we should fail fast as soon as there's a conflict, rather than trying to optimize the case where the conflict is resolved further down)